Framework/DataHandling/src/SaveDiffFittingAscii.cpp

```diff
@@ -0,0 +1,315 @@
+
+  const std::string filename = getProperty("Filename");
+  const std::string outMode = getProperty("OutMode");
+  std::string runNumList = getProperty("RunNumber");
+  std::string bankList = getProperty("Bank");
+
+  Poco::File pFile = (filename);
+  bool exist = pFile.exists();
+
+  bool appendToFile = false;
+  if (outMode == "AppendToExistingFile")
+    appendToFile = true;
+
+  // Initialize the file stream
+  std::ofstream file(filename.c_str(),
+                     (appendToFile ? std::ios::app : std::ios::out));
+
+  if (!file) {
+    throw Exception::FileError("Unable to create file: ", filename);
+  }
+
+  if (exist && !appendToFile) {
+    g_log.warning() << "File " << filename << " exists and will be overwritten."
+                    << "\n";
+  }
+
+  if (exist && appendToFile) {
+    file << "\n";
+  }
+
+  // reset counter
+  m_counter = 0;
+
+  std::vector<std::string> splitRunNum = splitList(runNumList);
+  std::vector<std::string> splitBank = splitList(bankList);
+
+  // Create a progress reporting object
+  Progress progress(this, 0, 1, input_ws.size());
+
+  size_t breaker = input_ws.size();
+  if (outMode == "AppendToExistingFile" && input_ws.size() == 1)
+    breaker = 1;
+
+  for (size_t i = 0; i < breaker; ++i) {
+
+    std::string runNum = splitRunNum[m_counter];
+    std::string bank = splitBank[m_counter];
+    writeInfo(runNum, bank, file);
+
+    // write header
+    std::vector<std::string> columnHeadings = input_ws[i]->getColumnNames();
+    writeHeader(columnHeadings, file);
+
+    // write out the data form the table workspace
+    size_t columnSize = columnHeadings.size();
+    writeData(input_ws[i], file, columnSize);
+
+    if (input_ws.size() > 1 && (i + 1) != input_ws.size()) {
+      file << '\n';
+    }
+  }
+  progress.report();
+}
+
+std::vector<std::string> SaveDiffFittingAscii::splitList(std::string strList) {
+  std::vector<std::string> splitList;
+  strList.erase(std::remove(strList.begin(), strList.end(), ' '),
+                strList.end());
+  boost::split(splitList, strList, boost::is_any_of(","));
+
+  return splitList;
+}
+
+void SaveDiffFittingAscii::writeInfo(const std::string &runNumber,
+                                     const std::string &bank,
+                                     std::ofstream &file) {
+
+  file << "run number: " << runNumber << '\n';
+  file << "bank: " << bank << '\n';
+  m_counter++;
+}
+
+void SaveDiffFittingAscii::writeHeader(std::vector<std::string> &columnHeadings,
+                                       std::ofstream &file) {
+  for (const auto &heading : columnHeadings) {
+    // Chi being the last header in the table workspace
+    if (heading == "Chi") {
+      writeVal(heading, file, true);
+    } else {
+      writeVal(heading, file, false);
+    }
+  }
+}
+
+void SaveDiffFittingAscii::writeData(API::ITableWorkspace_sptr workspace,
+                                     std::ofstream &file, size_t columnSize) {
```

Comment:
`const columnSize` 
This will work even with `(columnIndex == columnSize - 1)` as the compiler will evaluate is as a RHS val and not an assignment operator